service/collector/helper.go

```diff
@@ -183,3 +176,19 @@ func (h *helper) AWSAccountID(awsClients clientaws.Clients) (string, error) {
 
 	return accountID, nil
 }
+
+// ListReconciledClusters provides a list of clusters reconciled by this
+// particular operator version.
+func (h *helper) ListReconciledClusters() (*infrastructurev1alpha2.AWSClusterList, error) {
+	ctx := context.Background()
+
+	clusters := &infrastructurev1alpha2.AWSClusterList{}
+	err := h.clients.CtrlClient().List(
+		ctx,
+		clusters,
+		runtimeclient.MatchingLabels{
+			label.OperatorVersion: project.Version(),
+		},
+	)
+	return clusters, err
```

Comment:
Let's have masking and explicit error handling here. This is important for debugging because the stack then tells a story where the code is flowing and the consistency of the pattern makes it easier to work with the code.